### ISSUE TITLE
Multiple Method on a Route

### ISSUE THREAD

**Issue by pandacodez** [hapipal/discuss#3]
Hi,

I am new to hapi and hapipal, I am wondering if I can have multiple method on a route, hapi allows you do that right?, but when I add more that two methods, it return error when I am using hapipal with HauteCouture, I am using the boilerplate from hapipal

```
module.exports = {
    method: ['GET','POST'],
    path: '/abc',
    options: {
        handler:  (request, h) => {

            return ({ n: 'test', status: 'pending1' });

        }
    }
};
```

Error:

```
server/index.js:30
        throw err;
        ^

Error: server.route() called by haute using lib/routes/abc.js: Route id abc for path /abc conflicts with existing path /abc
    at new module.exports (node_modules/@hapi/hoek/lib/error.js:23:19)
    at Object.module.exports [as assert] (node_modules/@hapi/hoek/lib/assert.js:20:11)
    at exports.Router.internals.Router.internals.Router.add (node_modules/@hapi/call/lib/index.js:69:14)
    at internals.Server._addRoute (node_modules/@hapi/hapi/lib/server.js:489:46)
    at internals.Server.route (node_modules/@hapi/hapi/lib/server.js:474:26)
    at Object.internals.callMethod (node_modules/haute/lib/index.js:191:26)
    at makeCall (node_modules/haute/lib/index.js:174:25)
    at Object.exports.run (node_modules/haute/lib/index.js:178:15)
    at node_modules/haute-couture/lib/index.js:33:28
    at Object.register (lib/index.js:12:35)
[nodemon] app crashed - waiting for file changes before starting...
```

Second Questions, want to know why the handler is inside the options? the hapi put the handler same level as the method and path, is it because of HauteCouture

Why this, 
```
method: ['GET','POST'],
path: '/abc',
options: {
        handler:  (request, h) => {

            return ({ n: 'test', status: 'pending1' });

        }
}
```

instead of 
```
method: ['GET','POST'],
path: '/abc',
handler:  (request, h) => {
     return ({ n: 'test', status: 'pending1' });
}
```

**Comment by devinivy** [comment]
Good questions!

1. When you configure a route with two methods, behind the scenes hapi will create two routes that are identical but each have a single method.  At the same time, haute-couture is automatically configuring your route's [`id`](https://hapi.dev/api/?v=18.4.0#route.options.id) for you using the file name.  This causes hapi to create two routes with the same id, which fails.  One simple way to tell haute-couture to not configure the route with an `id` is to simply wrap it in an array:

   ```js
   module.exports = [{
       method: ['GET','POST'],
       path: '/abc',
       options: {
           handler:  (request, h) => {

               return ({ n: 'test', status: 'pending1' });

           }
       }
   }];
   ```

   It's also worth mentioning that hapi will not allow you to configure `route.options.validate.payload` on a route like this because `get` routes cannot have payload validation.  So you might run into another issue when you add validation to the route.  I would recommend making two routes, and sharing the configuration/handler logic between them in some other way.

   Fun fact!  You can use [hpal-debug](https://github.com/hapipal/hpal-debug) to hit your routes by id over the command line!
   ```
   hpal run debug:curl my-route-id
   ```

2. In general haute-couture passes the file contents directly through to a hapi server method (unless otherwise noted in [the docs](https://hapipal.com/docs/haute-couture#routes)).  For routes, haute-couture passes the contents of your file, a plain hapi route configuration, to `server.route()`.  hapi actually supports both styles `route.handler` and `route.options.handler`, which you can see here and here: https://hapi.dev/api/?v=18.4.0#-serverrouteroute https://hapi.dev/api/?v=18.4.0#route.options.handler.

**Comment by pandacodez** [comment]
Wow, you answered all my questions clearly, thank you so much. Thanks for the letting me know about the validation issue that I will face if I do it this way. Cheers.
